server/src/main/java/io/druid/metadata/SQLMetadataSegmentManager.java

```diff
@@ -161,7 +160,10 @@ public void stop()
     }
   }
 
-  private <T> T inReadOnlyTransaction(final TransactionCallback<T> callback)
+  protected static final <T> T inReadOnlyTransaction(
+      final SQLMetadataConnector connector,
```

Comment:
because I moved it to static. I don't think `IndexerSQLMetadataStorageCoordinator` connects to a `SQLMetadataSegmentManager`